daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/infoset/InfosetImpl.scala

```diff
@@ -213,7 +213,7 @@ case class InfosetValueLengthUnknownException(lengthState: LengthState, override
 final class FakeDINode extends DISimple(null) {
   private def die = throw new InfosetNoInfosetException(Nope)
 
-  override def parent = die
+  override val parent = die
```

Comment:
No idea how val parent = die works unless this val is overridden in all cases. Should be def.

Reply:
Maybe FakeDINode is just never allocated? Looks like it's only allocated when currentNode is called in DStateForConstantFolding. Maybe we just don't have a test for that?

Reply:
That is the only explanation. I've fixed the val by changing to def anyway.